psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -633,33 +635,28 @@ void HF::form_H() {
 
                             double jacobian = weight * r * r * sin(theta);
 
-                            basisset_->compute_phi(phi_ao, x, y, z);
+                            basisset_->compute_phi(phi_ao.pointer(), x, y, z);
 
-                            C_DGEMV('t', nao, nso, 1.0, &(u[0][0]), nso, &(phi_ao[0]), 1, 0.0, &(phi_so[0]), 1);
+                            phi_so.gemv(true, 1.0, &u, &phi_ao, 0.0);
 
                             for (int i = 0; i < nso; i++)
                                 for (int j = 0; j < nso; j++)
-                                    V_eff[i][j] += jacobian * (-1.0e6) * phi_so[i] * phi_so[j];
+                                    V_eff.add(i, j, jacobian * (-1.0e6) * phi_so[i] * phi_so[j]);
                         }
                     }
                 }
             }  // sphere
 
             outfile->Printf("  Perturbing H by %f %f %f V_eff.\n", dipole_field_strength_[0], dipole_field_strength_[1],
                             dipole_field_strength_[2]);
-            if (options_.get_int("PRINT") > 3) mat_print(V_eff, nso, nso, "outfile");
+            if (options_.get_int("PRINT") > 3) V_eff.print_out();
 
             if (dipole_field_type_ == dx) {
-                for (int i = 0; i < nso; i++)
-                    for (int j = 0; j < nso; j++) V_->set(i, j, V_eff[i][j]);  // ignore nuclear potential
+                V_->copy(V_eff);
```

Comment:
```suggestion
                V_->copy(V_eff); // ignore nuclear potential
```

Reply:
Is this not obvious from the code? Adding a comment felt redundant to me.

Reply:
perhaps. Not insisting on it.